cmdstanpy/utils/filesystem.py

```diff
@@ -183,12 +188,13 @@ def __init__(self, file_path: str):
             except RuntimeError:
                 pass
 
-        if ' ' in os.path.abspath(file_path):
+        if self.PATTERN.search(os.path.abspath(file_path)):
             tmpdir = tempfile.mkdtemp()
-            if ' ' in tmpdir:
+            if self.PATTERN.search(tmpdir):
                 raise RuntimeError(
-                    'Unable to generate temporary path without spaces! \n'
-                    + 'Please move your stan file to location without spaces.'
+                    'Unable to generate temporary path without special '
+                    'characters! \n Please move your stan file to location '
```

Comment:
I think we should say 'special characters or spaces' since the space is by far the most common issue but might not immediately be recognizable as special to users